Discovery/Discovery/Presentation/DiscoveryViewModel.swift

```diff
@@ -19,6 +19,7 @@ public class DiscoveryViewModel: ObservableObject {
     
     @Published var courses: [CourseItem] = []
     @Published var showError: Bool = false
+    @Published var searchQuery: String?
```

Comment:
@Published is redundant here because this variable does not affect UI changes.
Btw, the DiscoveryViewModel doesn't use this; maybe we can move it to the DiscoveryView?

Reply:
Done. moved to DiscoveryView.